src/app/WriteHandler.cpp

```diff
@@ -616,22 +616,29 @@ Status WriteHandler::ProcessWriteRequest(System::PacketBufferHandle && aPayload,
     return status;
 }
 
-CHIP_ERROR WriteHandler::AddStatus(const ConcreteDataAttributePath & aPath, const Protocols::InteractionModel::Status aStatus)
+CHIP_ERROR WriteHandler::AddStatus(const ConcreteDataAttributePath & aPath,
+                                   const Protocols::InteractionModel::ClusterStatusCode & aStatus)
 {
-    return AddStatus(aPath, StatusIB(aStatus));
+    if (aStatus.HasClusterSpecificCode())
+    {
+        return AddStatusInternal(
+            aPath, StatusIB{ (aStatus.IsSuccess() ? Status::Success : Status::Failure), aStatus.GetClusterSpecificCode().Value() });
```

Comment:
same comment here about why this would be legal. Since used in two places, maybe it would make sense to re-use code? We seem to generally want to get an StatusIB from a ClusterStatusCode.

Reply:
Done